src/NuGet.Core/NuGet.Protocol/Plugins/PluginDiscoverer.cs

```diff
@@ -17,25 +19,35 @@ public sealed class PluginDiscoverer : IPluginDiscoverer
     {
         private bool _isDisposed;
         private List<PluginFile> _pluginFiles;
-        private readonly string _rawPluginPaths;
+        private readonly string _envVariableNetCoreAndNetFXPluginPaths;
+        private readonly string _envVariableNuGetPluginPaths;
         private IEnumerable<PluginDiscoveryResult> _results;
         private readonly SemaphoreSlim _semaphore;
         private readonly EmbeddedSignatureVerifier _verifier;
+        private readonly IEnvironmentVariableReader _environmentVariableReader;
+
+        public PluginDiscoverer(string rawPluginPaths, EmbeddedSignatureVerifier verifier)
+            : this(rawPluginPaths, verifier, EnvironmentVariableWrapper.Instance)
+        {
+        }
 
         /// <summary>
         /// Instantiates a new <see cref="PluginDiscoverer" /> class.
         /// </summary>
         /// <param name="rawPluginPaths">The raw semicolon-delimited list of supposed plugin file paths.</param>
         /// <param name="verifier">An embedded signature verifier.</param>
+        /// <param name="environmentVariableReader"> A reader for environmental varibales. </param>
         /// <exception cref="ArgumentNullException">Thrown if <paramref name="verifier" /> is <see langword="null" />.</exception>
-        public PluginDiscoverer(string rawPluginPaths, EmbeddedSignatureVerifier verifier)
+        internal PluginDiscoverer(string rawPluginPaths, EmbeddedSignatureVerifier verifier, IEnvironmentVariableReader environmentVariableReader)
         {
             if (verifier == null)
             {
                 throw new ArgumentNullException(nameof(verifier));
             }
 
-            _rawPluginPaths = rawPluginPaths;
+            _environmentVariableReader = environmentVariableReader;
+            _envVariableNetCoreAndNetFXPluginPaths = rawPluginPaths;
+            _envVariableNuGetPluginPaths = _environmentVariableReader.GetEnvironmentVariable(EnvironmentVariableConstants.PluginPaths);
```

Comment:
I think we moved every logic out of the PluginManager that uses the env vars, so we probably shouldn't even be fetching the env vars from there. 
We can probably make the plugin discoverer take an IEnvironmentVariableReader and call it good. 
Especially when combined with the other change that's coming in from Andy that already breaks the public API.

Reply:
I was avoiding breaking the API, but it looks like it would be a cleaner implementation if we retrieve the env variables for plugins in PluginDiscoverer than PluginManager.